### ISSUE TITLE
Error after craft4 upgrade

### ISSUE THREAD

**Issue by xinningsu** [verbb/formie#1291]
### Describe the bug

Hi, need some help here. 
Recently I tried to upgrade Craftcms from Craft Pro 3.7.55.2 to Craft Pro 4.3.6.1, and upgrade Formie from 1.6.5 to 2.0.20. there is no error in migration, but there is an issue after upgrade:

There is a form footerJoinForm I can not submit or open in admin.
The error message as below, also the table structure:
```
[web.ERROR] [yii\db\Exception] SQLSTATE[42S22]: Column not found: 1054 Unknown column 'content.field_full_name_ilohqkwh' in 'field list' The SQL being executed was: SELECT EXISTS(SELECT `elements`.`id`, `elements`.`canonicalId`, `elements`.`fieldLayoutId`, `elements`.`uid`, `elements`.`enabled`, `elements`.`archived`, `elements`.`dateLastMerged`, `elements`.`dateCreated`, `elements`.`dateUpdated`, `elements_sites`.`id` AS `siteSettingsId`, `elements_sites`.`slug`, `elements_sites`.`siteId`, `elements_sites`.`uri`, `elements_sites`.`enabled` AS `enabledForSite`, `formie_submissions`.`id`, `formie_submissions`.`title`, `formie_submissions`.`formId`, `formie_submissions`.`statusId`, `formie_submissions`.`userId`, `formie_submissions`.`isIncomplete`, `formie_submissions`.`isSpam`, `formie_submissions`.`spamReason`, `formie_submissions`.`spamClass`, `formie_submissions`.`snapshot`, `formie_submissions`.`ipAddress`, `content`.`id` AS `contentId`, `content`.`field_full_name_ilohqkwh`, `content`.`field_email_zygzdljl`, `content`.`field_dateOfBirth_mrlsuxxt`, `content`.`field_location_chukogln` FROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId` FROM `elements` `elements` INNER JOIN `formie_submissions` `formie_submissions` ON `formie_submissions`.`id` = `elements`.`id` INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id` INNER JOIN `fmc_footerjoinform` `content` ON (`content`.`elementId` = `elements`.`id`) AND (`content`.`siteId` = `elements_sites`.`siteId`) WHERE (`formie_submissions`.`formId`='285995') AND (`formie_submissions`.`isIncomplete`=FALSE) AND (`formie_submissions`.`isSpam`=FALSE) AND (`elements_sites`.`siteId` IN (4, 6, 8, 7)) AND (`elements`.`archived`=FALSE) AND ((`elements`.`enabled`=TRUE) AND (`elements_sites`.`enabled`=TRUE)) AND (`elements`.`dateDeleted` IS NULL) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL) ORDER BY `elements`.`dateCreated` DESC LIMIT 1) `subquery` INNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId` INNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId` INNER JOIN `formie_submissions` `formie_submissions` ON `formie_submissions`.`id` = `subquery`.`elementsId` INNER JOIN `fmc_footerjoinform` `content` ON `content`.`id` = `subquery`.`contentId` ORDER BY `elements`.`dateCreated` DESC) {"trace":["#0 /app/vendor/yiisoft/yii2/db/Command.php(1307): yii\\db\\Schema->convertException()","#1 /app/vendor/yiisoft/yii2/db/Command.php(1168): yii\\db\\Command->internalExecute()","#2 /app/vendor/yiisoft/yii2/db/Command.php(436): yii\\db\\Command->queryInternal()","#3 /app/vendor/yiisoft/yii2/db/Query.php(446): yii\\db\\Command->queryScalar()","#4 /app/vendor/craftcms/cms/src/db/Query.php(321): yii\\db\\Query->exists()","#5 /app/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1540): craft\\db\\Query->exists()","#6 /app/vendor/twig/twig/src/Extension/CoreExtension.php(1607): craft\\elements\\db\\ElementQuery->exists()","#7 /app/vendor/craftcms/cms/src/helpers/Template.php(110): twig_get_attribute()","#8 /app/storage/cccsyd/runtime/compiled_templates/4b/4b85ab068aecc54d56379a7106b75a97c50f3a11ef1810c16fa82f6b80b33d26.php(85): craft\\helpers\\Template::attribute()","#9 /app/vendor/twig/twig/src/Template.php(171): __TwigTemplate_8dc3c507b0a9e260a9d861776bf2e08d0d534e86b12a83fec9de2037b145e372->block_actionButton()","#10 /app/vendor/twig/twig/src/Template.php(243): Twig\\Template->displayBlock()","#11 /app/storage/cccsyd/runtime/compiled_templates/cb/cb7d422f3f353a955f73c48a61aeabd96d702bc0160091632825274260acf162.php(101): Twig\\Template->renderBlock()","#12 /app/vendor/twig/twig/src/Template.php(394): __TwigTemplate_fb728922ad6c58c07167c964710045726718822ee910b7616756fad145f24840->doDisplay()","#13 /app/vendor/twig/twig/src/Template.php(367): Twig\\Template->displayWithErrorHandling()","#14 /app/storage/cccsyd/runtime/compiled_templates/4b/4b85ab068aecc54d56379a7106b75a97c50f3a11ef1810c16fa82f6b80b33d26.php(59): Twig\\Template->display()","#15 /app/vendor/twig/twig/src/Template.php(394): __TwigTemplate_8dc3c507b0a9e260a9d861776bf2e08d0d534e86b12a83fec9de2037b145e372->doDisplay()","#16 /app/vendor/twig/twig/src/Template.php(367): Twig\\Template->displayWithErrorHandling()","#17 /app/vendor/twig/twig/src/Template.php(379): Twig\\Template->display()","#18 /app/vendor/twig/twig/src/TemplateWrapper.php(40): Twig\\Template->render()","#19 /app/vendor/twig/twig/src/Environment.php(277): Twig\\TemplateWrapper->render()","#20 /app/vendor/craftcms/cms/src/web/View.php(451): Twig\\Environment->render()","#21 /app/vendor/craftcms/cms/src/web/View.php(504): craft\\web\\View->renderTemplate()","#22 /app/vendor/craftcms/cms/src/web/TemplateResponseFormatter.php(56): craft\\web\\View->renderPageTemplate()","#23 /app/vendor/yiisoft/yii2/web/Response.php(1098): craft\\web\\TemplateResponseFormatter->format()","#24 /app/vendor/craftcms/cms/src/web/Response.php(286): yii\\web\\Response->prepare()","#25 /app/vendor/yiisoft/yii2/web/Response.php(339): craft\\web\\Response->prepare()","#26 /app/vendor/yiisoft/yii2/base/Application.php(390): yii\\web\\Response->send()","#27 /app/web/index.php(84): yii\\base\\Application->run()","#28 {main}"],"memory":21672792,"exception":"[object] (yii\\db\\Exception(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'content.field_full_name_ilohqkwh' in 'field list'\nThe SQL being executed was: SELECT EXISTS(SELECT `elements`.`id`, `elements`.`canonicalId`, `elements`.`fieldLayoutId`, `elements`.`uid`, `elements`.`enabled`, `elements`.`archived`, `elements`.`dateLastMerged`, `elements`.`dateCreated`, `elements`.`dateUpdated`, `elements_sites`.`id` AS `siteSettingsId`, `elements_sites`.`slug`, `elements_sites`.`siteId`, `elements_sites`.`uri`, `elements_sites`.`enabled` AS `enabledForSite`, `formie_submissions`.`id`, `formie_submissions`.`title`, `formie_submissions`.`formId`, `formie_submissions`.`statusId`, `formie_submissions`.`userId`, `formie_submissions`.`isIncomplete`, `formie_submissions`.`isSpam`, `formie_submissions`.`spamReason`, `formie_submissions`.`spamClass`, `formie_submissions`.`snapshot`, `formie_submissions`.`ipAddress`, `content`.`id` AS `contentId`, `content`.`field_full_name_ilohqkwh`, `content`.`field_email_zygzdljl`, `content`.`field_dateOfBirth_mrlsuxxt`, `content`.`field_location_chukogln`\nFROM (SELECT `elements`.`id` AS `elementsId`, `elements_sites`.`id` AS `elementsSitesId`, `content`.`id` AS `contentId`\nFROM `elements` `elements`\nINNER JOIN `formie_submissions` `formie_submissions` ON `formie_submissions`.`id` = `elements`.`id`\nINNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`elementId` = `elements`.`id`\nINNER JOIN `fmc_footerjoinform` `content` ON (`content`.`elementId` = `elements`.`id`) AND (`content`.`siteId` = `elements_sites`.`siteId`)\nWHERE (`formie_submissions`.`formId`='285995') AND (`formie_submissions`.`isIncomplete`=FALSE) AND (`formie_submissions`.`isSpam`=FALSE) AND (`elements_sites`.`siteId` IN (4, 6, 8, 7)) AND (`elements`.`archived`=FALSE) AND ((`elements`.`enabled`=TRUE) AND (`elements_sites`.`enabled`=TRUE)) AND (`elements`.`dateDeleted` IS NULL) AND (`elements`.`draftId` IS NULL) AND (`elements`.`revisionId` IS NULL)\nORDER BY `elements`.`dateCreated` DESC\nLIMIT 1) `subquery`\nINNER JOIN `elements` `elements` ON `elements`.`id` = `subquery`.`elementsId`\nINNER JOIN `elements_sites` `elements_sites` ON `elements_sites`.`id` = `subquery`.`elementsSitesId`\nINNER JOIN `formie_submissions` `formie_submissions` ON `formie_submissions`.`id` = `subquery`.`elementsId`\nINNER JOIN `fmc_footerjoinform` `content` ON `content`.`id` = `subquery`.`contentId`\nORDER BY `elements`.`dateCreated` DESC) at /app/vendor/yiisoft/yii2/db/Schema.php:676)\n[previous exception] [object] (PDOException(code: 42S22): SQLSTATE[42S22]: Column not found: 1054 Unknown column 'content.field_full_name_ilohqkwh' in 'field list' at /app/vendor/yiisoft/yii2/db/Command.php:1302)"} 
```
Table structure before upgrade:
```
CREATE TABLE `fmc_footerjoinform` (
  `id` int NOT NULL AUTO_INCREMENT,
  `elementId` int NOT NULL,
  `siteId` int NOT NULL,
  `dateCreated` datetime NOT NULL,
  `dateUpdated` datetime NOT NULL,
  `uid` char(36) NOT NULL DEFAULT '0',
  `field_full_name_ilohqkwh` varchar(255) DEFAULT NULL,
  `field_email_zygzdljl` varchar(255) DEFAULT NULL,
  `field_location_chukogln` varchar(255) DEFAULT NULL,
  `field_dateOfBirth_mrlsuxxt` datetime DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `idx_jxheodzvirkeqxbyigbyzeddultrqhmnjitz` (`elementId`,`siteId`),
  KEY `idx_xmprojkgzrtcvzaagtblziszmeadqwvabetu` (`siteId`),
  CONSTRAINT `fk_dtqlmtocfbsyzwkswynjdkfolwbpxfksywqj` FOREIGN KEY (`elementId`) REFERENCES `elements` (`id`) ON DELETE CASCADE,
  CONSTRAINT `fk_uoleyaviceuirtpzlqlovnubateehjzjvcgr` FOREIGN KEY (`siteId`) REFERENCES `sites` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3;
```
Table structure after upgrade:
```
Create Table: CREATE TABLE `fmc_footerjoinform` (
  `id` int NOT NULL AUTO_INCREMENT,
  `elementId` int NOT NULL,
  `siteId` int NOT NULL,
  `dateCreated` datetime NOT NULL,
  `dateUpdated` datetime NOT NULL,
  `uid` char(36) NOT NULL DEFAULT '0',
  `field_email_zygzdljl` varchar(255) DEFAULT NULL,
  `field_location_chukogln` varchar(255) DEFAULT NULL,
  `field_dateOfBirth_mrlsuxxt` datetime DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `idx_jxheodzvirkeqxbyigbyzeddultrqhmnjitz` (`elementId`,`siteId`),
  KEY `idx_xmprojkgzrtcvzaagtblziszmeadqwvabetu` (`siteId`),
  CONSTRAINT `fk_dtqlmtocfbsyzwkswynjdkfolwbpxfksywqj` FOREIGN KEY (`elementId`) REFERENCES `elements` (`id`) ON DELETE CASCADE,
  CONSTRAINT `fk_uoleyaviceuirtpzlqlovnubateehjzjvcgr` FOREIGN KEY (`siteId`) REFERENCES `sites` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb3
```

### Steps to reproduce

1. Upgrade to craft 4
2. Edit a form

### Form settings

- Multi-page form: Yes
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: No

### Craft CMS version

4.3.6.1

### Plugin version

2.0.20

### Multi-site?

Yes

**Comment by engram-design** [comment]
Does adding the `field_full_name_ilohqkwh` column back manually fix things? Is this this only form with that issue?

Can you comment on what sort of field the `full_name` field is?

It's very likely this was fixed in [2.0.21](https://github.com/verbb/formie/blob/craft-4/CHANGELOG.md#2021---2023-01-30), notably:

> Fixed a migration error when pruning synced fields, where the field handle contains underscores.

And I note that your field handle contains an underscore.

**Comment by xinningsu** [comment]
The problem has been solved after upgrade formie to 2.0.22
